// curseforge-api/v1/File.js
import CurseForgeBase from './Base.js';
export default class CurseForgeFile extends CurseForgeBase {
    id;
    gameId;
    modId;
    isAvailable;
    displayName;
    fileName;
    releaseType;
    fileStatus;
    hashes;
    fileDate;
    fileLength;
    downloadCount;
    fileSizeOnDisk;
    downloadUrl;
    gameVersions;
    sortableGameVersions;
    dependencies;
    exposeAsAlternative;
    parentProjectFileId;
    alternateFileId;
    isServerPack;
    serverPackFileId;
    isEarlyAccessContent;
    earlyAccessEndDate;
    fileFingerprint;
    modules;
    constructor(client, data) {
        super(client);
        this.id = data.id;
        this.gameId = data.gameId;
        this.modId = data.modId;
        this.isAvailable = data.isAvailable;
        this.displayName = data.displayName;
        this.fileName = data.fileName;
        this.releaseType = data.releaseType;
        this.fileStatus = data.fileStatus;
        this.hashes = data.hashes;
        this.fileDate = data.fileDate;
        this.fileLength = data.fileLength;
        this.downloadCount = data.downloadCount;
        this.fileSizeOnDisk = data.fileSizeOnDisk;
        this.downloadUrl = data.downloadUrl;
        this.gameVersions = data.gameVersions;
        this.sortableGameVersions = data.sortableGameVersions;
        this.dependencies = data.dependencies;
        this.exposeAsAlternative = data.exposeAsAlternative;
        this.parentProjectFileId = data.parentProjectFileId;
        this.alternateFileId = data.alternateFileId;
        this.isServerPack = data.isServerPack;
        this.serverPackFileId = data.serverPackFileId;
        this.isEarlyAccessContent = data.isEarlyAccessContent;
        this.earlyAccessEndDate = data.earlyAccessEndDate;
        this.fileFingerprint = data.fileFingerprint;
        this.modules = data.modules;
    }
    getChangelog() {
        return this.client.getModFileChangelog(this.modId, this.id);
    }
    getDownloadURL() {
        return this.client.getModFileDownloadURL(this.modId, this.id);
    }
}
