// curseforge-api/v1/GameEnum.js
var CurseForgeGameEnum;
(function (CurseForgeGameEnum) {
    CurseForgeGameEnum[CurseForgeGameEnum["WorldofWarcraft"] = 1] = "WorldofWarcraft";
    CurseForgeGameEnum[CurseForgeGameEnum["TheSecretWorld"] = 64] = "TheSecretWorld";
    CurseForgeGameEnum[CurseForgeGameEnum["StarCraftII"] = 65] = "StarCraftII";
    CurseForgeGameEnum[CurseForgeGameEnum["RunesofMagic"] = 335] = "RunesofMagic";
    CurseForgeGameEnum[CurseForgeGameEnum["WorldofTanks"] = 423] = "WorldofTanks";
    CurseForgeGameEnum[CurseForgeGameEnum["Rift"] = 424] = "Rift";
    CurseForgeGameEnum[CurseForgeGameEnum["Terraria"] = 431] = "Terraria";
    CurseForgeGameEnum[CurseForgeGameEnum["Minecraft"] = 432] = "Minecraft";
    CurseForgeGameEnum[CurseForgeGameEnum["WildStar"] = 454] = "WildStar";
    CurseForgeGameEnum[CurseForgeGameEnum["TheElderScrollsOnline"] = 455] = "TheElderScrollsOnline";
    CurseForgeGameEnum[CurseForgeGameEnum["DarkestDungeon"] = 608] = "DarkestDungeon";
    CurseForgeGameEnum[CurseForgeGameEnum["StardewValley"] = 669] = "StardewValley";
    CurseForgeGameEnum[CurseForgeGameEnum["SidMeiersCivilizationVI"] = 727] = "SidMeiersCivilizationVI";
    CurseForgeGameEnum[CurseForgeGameEnum["KerbalSpaceProgram"] = 4401] = "KerbalSpaceProgram";
    CurseForgeGameEnum[CurseForgeGameEnum["SecretWorldLegends"] = 4455] = "SecretWorldLegends";
    CurseForgeGameEnum[CurseForgeGameEnum["FinalFantasyIV"] = 4741] = "FinalFantasyIV";
    CurseForgeGameEnum[CurseForgeGameEnum["FinalFantasyVI"] = 4773] = "FinalFantasyVI";
    CurseForgeGameEnum[CurseForgeGameEnum["FinalFantasyII"] = 5001] = "FinalFantasyII";
    CurseForgeGameEnum[CurseForgeGameEnum["FinalFantasyV"] = 5021] = "FinalFantasyV";
    CurseForgeGameEnum[CurseForgeGameEnum["FinalFantasyIII"] = 5026] = "FinalFantasyIII";
    CurseForgeGameEnum[CurseForgeGameEnum["FinalFantasyI"] = 5230] = "FinalFantasyI";
    CurseForgeGameEnum[CurseForgeGameEnum["SurvivingMars"] = 61489] = "SurvivingMars";
    CurseForgeGameEnum[CurseForgeGameEnum["MinecraftDungeons"] = 69271] = "MinecraftDungeons";
    CurseForgeGameEnum[CurseForgeGameEnum["AmongUs"] = 69761] = "AmongUs";
    CurseForgeGameEnum[CurseForgeGameEnum["ChroniclesofArcadia"] = 70667] = "ChroniclesofArcadia";
    CurseForgeGameEnum[CurseForgeGameEnum["TheSims4"] = 78062] = "TheSims4";
    CurseForgeGameEnum[CurseForgeGameEnum["Demeo"] = 78135] = "Demeo";
    CurseForgeGameEnum[CurseForgeGameEnum["RushdownRevolt"] = 84062] = "RushdownRevolt";
    CurseForgeGameEnum[CurseForgeGameEnum["TennisElbow4"] = 84137] = "TennisElbow4";
    CurseForgeGameEnum[CurseForgeGameEnum["Palworld"] = 85196] = "Palworld";
    CurseForgeGameEnum[CurseForgeGameEnum["Helldivers2"] = 85440] = "Helldivers2";
})(CurseForgeGameEnum || (CurseForgeGameEnum = {}));
export default CurseForgeGameEnum;
