// curseforge-api/v1/Types.js
export var CurseForgeCoreApiStatus;
(function (CurseForgeCoreApiStatus) {
    CurseForgeCoreApiStatus[CurseForgeCoreApiStatus["Private"] = 1] = "Private";
    CurseForgeCoreApiStatus[CurseForgeCoreApiStatus["Public"] = 2] = "Public";
})(CurseForgeCoreApiStatus || (CurseForgeCoreApiStatus = {}));
export var CurseForgeCoreStatus;
(function (CurseForgeCoreStatus) {
    CurseForgeCoreStatus[CurseForgeCoreStatus["Draft"] = 1] = "Draft";
    CurseForgeCoreStatus[CurseForgeCoreStatus["Test"] = 2] = "Test";
    CurseForgeCoreStatus[CurseForgeCoreStatus["PendingReview"] = 3] = "PendingReview";
    CurseForgeCoreStatus[CurseForgeCoreStatus["Rejected"] = 4] = "Rejected";
    CurseForgeCoreStatus[CurseForgeCoreStatus["Approved"] = 5] = "Approved";
    CurseForgeCoreStatus[CurseForgeCoreStatus["Live"] = 6] = "Live";
})(CurseForgeCoreStatus || (CurseForgeCoreStatus = {}));
export var CurseForgeFileRelationType;
(function (CurseForgeFileRelationType) {
    CurseForgeFileRelationType[CurseForgeFileRelationType["EmbeddedLibrary"] = 1] = "EmbeddedLibrary";
    CurseForgeFileRelationType[CurseForgeFileRelationType["OptionalDependency"] = 2] = "OptionalDependency";
    CurseForgeFileRelationType[CurseForgeFileRelationType["RequiredDependency"] = 3] = "RequiredDependency";
    CurseForgeFileRelationType[CurseForgeFileRelationType["Tool"] = 4] = "Tool";
    CurseForgeFileRelationType[CurseForgeFileRelationType["Incompatible"] = 5] = "Incompatible";
    CurseForgeFileRelationType[CurseForgeFileRelationType["Include"] = 6] = "Include";
})(CurseForgeFileRelationType || (CurseForgeFileRelationType = {}));
export var CurseForgeFileReleaseType;
(function (CurseForgeFileReleaseType) {
    CurseForgeFileReleaseType[CurseForgeFileReleaseType["Release"] = 1] = "Release";
    CurseForgeFileReleaseType[CurseForgeFileReleaseType["Beta"] = 2] = "Beta";
    CurseForgeFileReleaseType[CurseForgeFileReleaseType["Alpha"] = 3] = "Alpha";
})(CurseForgeFileReleaseType || (CurseForgeFileReleaseType = {}));
export var CurseForgeFileStatus;
(function (CurseForgeFileStatus) {
    CurseForgeFileStatus[CurseForgeFileStatus["Processing"] = 1] = "Processing";
    CurseForgeFileStatus[CurseForgeFileStatus["ChangesRequired"] = 2] = "ChangesRequired";
    CurseForgeFileStatus[CurseForgeFileStatus["UnderReview"] = 3] = "UnderReview";
    CurseForgeFileStatus[CurseForgeFileStatus["Approved"] = 4] = "Approved";
    CurseForgeFileStatus[CurseForgeFileStatus["Rejected"] = 5] = "Rejected";
    CurseForgeFileStatus[CurseForgeFileStatus["MalwareDetected"] = 6] = "MalwareDetected";
    CurseForgeFileStatus[CurseForgeFileStatus["Deleted"] = 7] = "Deleted";
    CurseForgeFileStatus[CurseForgeFileStatus["Archived"] = 8] = "Archived";
    CurseForgeFileStatus[CurseForgeFileStatus["Testing"] = 9] = "Testing";
    CurseForgeFileStatus[CurseForgeFileStatus["Released"] = 10] = "Released";
    CurseForgeFileStatus[CurseForgeFileStatus["ReadyForReview"] = 11] = "ReadyForReview";
    CurseForgeFileStatus[CurseForgeFileStatus["Deprecated"] = 12] = "Deprecated";
    CurseForgeFileStatus[CurseForgeFileStatus["Baking"] = 13] = "Baking";
    CurseForgeFileStatus[CurseForgeFileStatus["AwaitingPublishing"] = 14] = "AwaitingPublishing";
    CurseForgeFileStatus[CurseForgeFileStatus["FailedPublishing"] = 15] = "FailedPublishing";
})(CurseForgeFileStatus || (CurseForgeFileStatus = {}));
export var CurseForgeGameVersionStatus;
(function (CurseForgeGameVersionStatus) {
    CurseForgeGameVersionStatus[CurseForgeGameVersionStatus["Approved"] = 1] = "Approved";
    CurseForgeGameVersionStatus[CurseForgeGameVersionStatus["Deleted"] = 2] = "Deleted";
    CurseForgeGameVersionStatus[CurseForgeGameVersionStatus["New"] = 3] = "New";
})(CurseForgeGameVersionStatus || (CurseForgeGameVersionStatus = {}));
export var CurseForgeGameVersionTypeStatus;
(function (CurseForgeGameVersionTypeStatus) {
    CurseForgeGameVersionTypeStatus[CurseForgeGameVersionTypeStatus["Normal"] = 1] = "Normal";
    CurseForgeGameVersionTypeStatus[CurseForgeGameVersionTypeStatus["Deleted"] = 2] = "Deleted";
})(CurseForgeGameVersionTypeStatus || (CurseForgeGameVersionTypeStatus = {}));
export var CurseForgeHashAlgo;
(function (CurseForgeHashAlgo) {
    CurseForgeHashAlgo[CurseForgeHashAlgo["Sha1"] = 1] = "Sha1";
    CurseForgeHashAlgo[CurseForgeHashAlgo["Md5"] = 2] = "Md5";
})(CurseForgeHashAlgo || (CurseForgeHashAlgo = {}));
export var CurseForgeModLoaderInstallMethod;
(function (CurseForgeModLoaderInstallMethod) {
    CurseForgeModLoaderInstallMethod[CurseForgeModLoaderInstallMethod["ForgeInstaller"] = 1] = "ForgeInstaller";
    CurseForgeModLoaderInstallMethod[CurseForgeModLoaderInstallMethod["ForgeJarInstall"] = 2] = "ForgeJarInstall";
    CurseForgeModLoaderInstallMethod[CurseForgeModLoaderInstallMethod["ForgeInstaller_v2"] = 3] = "ForgeInstaller_v2";
})(CurseForgeModLoaderInstallMethod || (CurseForgeModLoaderInstallMethod = {}));
export var CurseForgeModLoaderType;
(function (CurseForgeModLoaderType) {
    CurseForgeModLoaderType[CurseForgeModLoaderType["Any"] = 0] = "Any";
    CurseForgeModLoaderType[CurseForgeModLoaderType["Forge"] = 1] = "Forge";
    CurseForgeModLoaderType[CurseForgeModLoaderType["Cauldron"] = 2] = "Cauldron";
    CurseForgeModLoaderType[CurseForgeModLoaderType["LiteLoader"] = 3] = "LiteLoader";
    CurseForgeModLoaderType[CurseForgeModLoaderType["Fabric"] = 4] = "Fabric";
    CurseForgeModLoaderType[CurseForgeModLoaderType["Quilt"] = 5] = "Quilt";
    CurseForgeModLoaderType[CurseForgeModLoaderType["NeoForge"] = 6] = "NeoForge";
})(CurseForgeModLoaderType || (CurseForgeModLoaderType = {}));
export var CurseForgeModsSearchSortField;
(function (CurseForgeModsSearchSortField) {
    CurseForgeModsSearchSortField[CurseForgeModsSearchSortField["Featured"] = 1] = "Featured";
    CurseForgeModsSearchSortField[CurseForgeModsSearchSortField["Popularity"] = 2] = "Popularity";
    CurseForgeModsSearchSortField[CurseForgeModsSearchSortField["LastUpdated"] = 3] = "LastUpdated";
    CurseForgeModsSearchSortField[CurseForgeModsSearchSortField["Name"] = 4] = "Name";
    CurseForgeModsSearchSortField[CurseForgeModsSearchSortField["Author"] = 5] = "Author";
    CurseForgeModsSearchSortField[CurseForgeModsSearchSortField["TotalDownloads"] = 6] = "TotalDownloads";
    CurseForgeModsSearchSortField[CurseForgeModsSearchSortField["Category"] = 7] = "Category";
    CurseForgeModsSearchSortField[CurseForgeModsSearchSortField["GameVersion"] = 8] = "GameVersion";
    CurseForgeModsSearchSortField[CurseForgeModsSearchSortField["EarlyAccess"] = 9] = "EarlyAccess";
    CurseForgeModsSearchSortField[CurseForgeModsSearchSortField["FeaturedReleased"] = 10] = "FeaturedReleased";
    CurseForgeModsSearchSortField[CurseForgeModsSearchSortField["ReleasedDate"] = 11] = "ReleasedDate";
    CurseForgeModsSearchSortField[CurseForgeModsSearchSortField["Rating"] = 12] = "Rating";
})(CurseForgeModsSearchSortField || (CurseForgeModsSearchSortField = {}));
export var CurseForgeModStatus;
(function (CurseForgeModStatus) {
    CurseForgeModStatus[CurseForgeModStatus["New"] = 1] = "New";
    CurseForgeModStatus[CurseForgeModStatus["ChangesRequired"] = 2] = "ChangesRequired";
    CurseForgeModStatus[CurseForgeModStatus["UnderSoftReview"] = 3] = "UnderSoftReview";
    CurseForgeModStatus[CurseForgeModStatus["Approved"] = 4] = "Approved";
    CurseForgeModStatus[CurseForgeModStatus["Rejected"] = 5] = "Rejected";
    CurseForgeModStatus[CurseForgeModStatus["ChangesMade"] = 6] = "ChangesMade";
    CurseForgeModStatus[CurseForgeModStatus["Inactive"] = 7] = "Inactive";
    CurseForgeModStatus[CurseForgeModStatus["Abandoned"] = 8] = "Abandoned";
    CurseForgeModStatus[CurseForgeModStatus["Deleted"] = 9] = "Deleted";
    CurseForgeModStatus[CurseForgeModStatus["UnderReview"] = 10] = "UnderReview";
})(CurseForgeModStatus || (CurseForgeModStatus = {}));
export var CurseForgeSortOrder;
(function (CurseForgeSortOrder) {
    CurseForgeSortOrder["Ascending"] = "asc";
    CurseForgeSortOrder["Descending"] = "desc";
})(CurseForgeSortOrder || (CurseForgeSortOrder = {}));
